javaslang/src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -51,6 +51,88 @@
         PUT, REMOVE
     }
 
+    class LeafNodeIterator<K, V> extends AbstractIterator<LeafNode<K, V>> {
+
+        // buckets levels + leaf level = (Integer.SIZE / AbstractNode.SIZE + 1) + 1
+        private final static int MAX_LEVELS = Integer.SIZE / AbstractNode.SIZE + 2;
```

Comment:
> ok, then MAX_LEVELS is an upper bound, right?

yes, just maximum value of tree depth

> why not set MAX_LEVELS to (Integer.SIZE / AbstractNode.SIZE + 1) + 1?

hmmm... I see no difference, but I can do it